### PR TITLE
[uncrustify/0.77.1] Bump recipe

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **lib/1.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
Yes, there are 2 way to fix this in test package for conan v2 client:
- replace `self.requires(self.tested_reference_str)` by `self.requires(self.tested_reference_str, run=True)`
- replace  `"VirtualRunEnv"` by `"VirtualBuildEnv"`, add `self.tested_reference_str` to `tool_requires` of `build_requirements()` instead of `requires` of `requirements()`, and in test() run `self.run("uncrustify --version")` unconditionally.

**Comment by MartinDelille** [comment]
That make perfectly  sense! Thank you!

File: recipes/uncrustify/all/test_package/conanfile.py [hunk 1]
```diff
@@ -4,12 +4,11 @@
...
 
-    def requirements(self):
-        self.requires(self.tested_reference_str)
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
 
     def test(self):
-        if can_run(self):
-            self.run("uncrustify --version", env="conanrun")
+        self.run("uncrustify --version", env="conanrun")
```
Comment (SpaceIm) [inline recipes/uncrustify/all/test_package/conanfile.py:14]:
```suggestion
        self.run("uncrustify --version")
```

**Comment by SpaceIm** [comment]
There is also an unused import of can_run to remove now.

**Review by SpaceIm — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**